_posts/blog/2016-07-07-reactive_mindset_burgers.md

```diff
@@ -0,0 +1,165 @@
+</picture>
+</center>
+
+The next step is to make the burger. Donald needs to take the first bun given by Huey, together with the first piece of cooked meat and with the first slice of tomato given by Louie and then "zip" together the first burger.
+
+For the second burger, he will use the second bun from Huey, the second piece of cooked meat and the second slice of tomato.The time took by each of the nephews to do his part is different. Louie will be providing tomato slices faster than Dewey can cook the meat. As a result, Donald has to wait for it to be ready, in order to make the burger. Because he doesn't want to throw away any tomato slices, he will always use the first one on the row, given by Louie.
+
+<center>
+<picture>
+	<a href="/images/blog/reactive_mindset_burgers/zip_burgers.png"><img src="/images/blog/reactive_mindset_burgers/zip_burgers.png" alt="zip burgers" width="350"></a>
+	<figcaption>Combine the bun, the meat and the slice of tomato to make the burger</figcaption>
+</picture>
+</center>
+
+## ReactiveX and Buns, Meat & Tomato Slices
+
+In ReactiveX the buns are streams of data. The meat and the tomato slices are also streams of data, and the burgers also, but created by composing multiple streams of data. The class that emits a stream of data is called ``Observable``. So this means that we will have an Observable of buns, an Observable of meat, an Observable of tomato slices and in the end, an Observable of burgers.
+
+<center>
+<picture>
+	<a href="/images/blog/reactive_mindset_burgers/streams_of_data.png"><img src="/images/blog/reactive_mindset_burgers/streams_of_data.png" alt="streams of data" width="350"></a>
+	<figcaption>The buns, pieces of meat, slices of tomato and the burgers are streams of data</figcaption>
+</picture>
+</center>
+
+Donald Duck's customer, Mickey Mouse, he gets to consume the burger. So, Mickey Mouse is the one that acts upon the emitted items by the burger Observable. In ReactiveX, this is the ``Subscriber`` class.
+
+The actions that we performed on the stream of meat given by Dewey: filtering the meat that has gone bad and cooking the meat - so applying a function to every piece of meat - are called ``operators``.
+
+When developing using ReactiveX you have to switch from the pull based approach to a push based one. More precisely, Donald Duck does not pull buns from Huey. He does not request buns. Rather, Donald is just waiting for buns that are delivered to him by his nephew.
+
+<center>
+<picture>
+	<a href="/images/blog/reactive_mindset_burgers/pull_bun.png"><img src="/images/blog/reactive_mindset_burgers/pull_bun.png" alt="pull based development" width="350"></a>
+	<figcaption>Donald Duck doesn't request buns from Huey</figcaption>
+</picture>
+</center>
+
+We don't have an exact set of data that we can operate on, but rather we operate on each emission of a stream. This means that if we want to work only on meat that has not gone bad, we cannot do:
```

Comment:
too much negations - This means that if we want to work just with good meat, we cannot do: